### PR TITLE
Answers

### PR DESCRIPTION
**Pull request by ahmednawar**
@sojanpr @aaustin 

You may also use the framework in a demo app directly for a quick live test.

### CODE REVIEW

File: Branch-TestBed/Branch-TestBed/ViewController.m [hunk 1]
```diff
@@ -52,17 +53,25 @@ - (IBAction)cmdRefreshShort:(id)sender {
 - (IBAction)cmdShareLink:(id)sender {
     BranchLinkProperties *linkProperties = [[BranchLinkProperties alloc] init];
     linkProperties.feature = @"sharing";
+    linkProperties.alias = @"some alias";
+    linkProperties.tags = @[@"tag1", @"tag2"];
+    linkProperties.matchDuration = 123;
```
Comment (aaustin) [inline Branch-TestBed/Branch-TestBed/ViewController.m]: nit: but can you make this 6000 or something so people don't think matching is broken while testing with our testbed.

File: Branch-TestBed/ANSCompatibility.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
```
Comment (aaustin) [inline Branch-TestBed/ANSCompatibility.h]: do we need these files in the testbed? or were you just testing if there'd be a conflict?

File: Branch-TestBed/Answers.h [hunk 3]
```diff
@@ -0,0 +1,237 @@
+//
```
Comment (aaustin) [inline Branch-TestBed/Answers.h]: same here

File: Branch-SDK/Branch-SDK/BNCFabricAnswers.m [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//
+
+#import "BNCFabricAnswers.h"
+#import "Answers.h"
+
+@implementation BNCFabricAnswers
+
++ (void)sendEventWithName:(NSString *)name andAttributes:(NSDictionary *)attributes {
+    ANSLogCustomEvent(name, [self prepareBranchDataForAnswers:attributes]);
+
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCFabricAnswers.m]: slight nit on extra new line spacing here

File: Branch-SDK/Branch-SDK/BNCFabricAnswers.m [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    for (NSString *key in dictionary.allKeys) {
+        if ([key hasPrefix:@"+"]) {
+            // ignore because this data is not found on the ShareSheet
+            continue;
+        } else if ([key hasPrefix:@"~"]) {
+            // strip tildes ~
+            temp[[key substringFromIndex:1]] = dictionary[key];
+        } else {
+            // link data
+            temp[[NSString stringWithFormat:@"data.%@", key]] = dictionary[key];
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCFabricAnswers.m]: Let's just add a quick check to make sure it's type NSNumber or NSString

File: Branch-SDK/Branch-SDK/BNCFabricAnswers.m [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        
+        //flatten arrays, separate if statement because they are caught in one of the prefix conditionals
+        if ([dictionary[key] isKindOfClass:[NSArray class]]) {
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCFabricAnswers.m]: maybe, just for consistency, move this into an `else if` before the `} else if ([key hasPrefix:@"~"]) {` above.

File: Branch-SDK/Branch-SDK/BranchUniversalObject.m [hunk 7]
```diff
@@ -164,6 +165,11 @@ - (void)showShareSheetWithLinkProperties:(BranchLinkProperties *)linkProperties
         shareViewController.completionWithItemsHandler = ^(NSString *activityType, BOOL completed, NSArray *returnedItems, NSError *activityError) {
             if (completion) {
                 completion(activityType, completed);
+                // DEBUG ONLY
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BranchUniversalObject.m]: can you just add a `#warning DEBUG ONLY` instead so we don't miss this

File: Branch-SDK/Branch-SDK/BNCFabricAnswers.m (hunk 5 above)
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/BNCFabricAnswers.m]: Wouldn't this fail for tags and keywords because they are arrays?

Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCFabricAnswers.m]: if you move the array check above the other else if, like I suggest below, then no.

File: Branch-TestBed/ANSCompatibility.h (hunk 2 above)
Comment (ahmednawar) [inline Branch-TestBed/ANSCompatibility.h]: I think we need them but I should move them to the Fabric subdirectory under Branch-SDK

**Comment by aaustin** [comment]
👍

File: Branch-SDK/Branch-SDK/BranchUniversalObject.m [hunk 8]
```diff
@@ -164,6 +165,12 @@ - (void)showShareSheetWithLinkProperties:(BranchLinkProperties *)linkProperties
         shareViewController.completionWithItemsHandler = ^(NSString *activityType, BOOL completed, NSArray *returnedItems, NSError *activityError) {
             if (completion) {
                 completion(activityType, completed);
+
+#warning DEBUG_ONLY
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BranchUniversalObject.m]: let's clean this up and test with prod backend!

**Comment by aaustin** [comment]
permission granted to merge and deploy after testing with the server flag

**Comment by aaustin** [comment]
👍

**Pull request merged**